type-tests/@ember/template-compilation.ts

```diff
@@ -0,0 +1,28 @@
+import { precompileTemplate } from '@ember/template-compilation';
+import { setComponentTemplate } from '@ember/component';
+import templateOnly from '@ember/component/template-only';
+
+import { expectTypeOf } from 'expect-type';
+
+// Valid usages
+precompileTemplate(`Hello World`, { moduleName: 'foo' });
+precompileTemplate(`Hello World`, { strictMode: true });
+precompileTemplate(`Hello World`, { strictMode: false });
+precompileTemplate(`Hello World`, { strictMode: true, scope: () => ({}) });
+precompileTemplate(`Hello World`, { strictMode: true, scope: () => ({ setComponentTemplate }) });
+precompileTemplate(`Hello World`, { strictMode: true, moduleName: 'hello', scope: () => ({}) });
+
+// Integration, since this is the primary use case for precompileTemplate
+expectTypeOf(setComponentTemplate(precompileTemplate(`Hello World`), templateOnly())).toBeObject();
```

Comment:
is there any value in keeping expectTypeOf here since setComponentTemplate is already tested?

Reply:
Seems fine to keep it here as well.